keycloak/main.tf

```diff
@@ -0,0 +1,197 @@
+  metadata {
+    name = var.namespace_name
+  }
+}
+
+resource "kubernetes_secret" "keycloak_auth" {
+  depends_on = [
+    var.module_depends_on
+  ]
+
+  count = var.keycloak_google_auth ? 1 - local.argocd_enabled : 0
+
+  metadata {
+    name      = "keycloak-auth"
+    namespace = local.namespace
+  }
+
+  data = {
+    KC_AUTH_GOOGLE_CLIENT_ID     = var.keycloak_client_id
+    KC_AUTH_GOOGLE_CLIENT_SECRET = var.keycloak_client_secret
+  }
+}
+
+resource "aws_kms_ciphertext" "keycloak_client_secret" {
+  count     = var.keycloak_google_auth && local.argocd_enabled > 0 ? 1 : 0
+  key_id    = var.argocd.kms_key_id
+  plaintext = base64encode(var.keycloak_client_secret)
+}
+
+resource "aws_kms_ciphertext" "keycloak_password" {
+  count     = local.argocd_enabled
+  key_id    = var.argocd.kms_key_id
+  plaintext = local.password
+}
+
+resource "local_file" "namespace" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  content = yamlencode({
+    "apiVersion" = "v1"
+    "kind"       = "Namespace"
+    "metadata" = {
+      "name" = local.namespace
+    }
+  })
+  filename = "${path.root}/${var.argocd.path}/ns-${local.namespace}.yaml"
+}
+
+resource "local_file" "keycloak_auth" {
+  count = var.keycloak_google_auth ? local.argocd_enabled : 0
+  depends_on = [
+    var.module_depends_on
+  ]
+  content = yamlencode({
+    "apiVersion" = "v1"
+    "kind"       = "Secret"
+    "metadata" = {
+      "name"      = "keycloak-auth"
+      "namespace" = local.namespace
+    }
+    "stringData" = {
+      "KC_AUTH_GOOGLE_CLIENT_ID"     = var.keycloak_client_id
+      "KC_AUTH_GOOGLE_CLIENT_SECRET" = "KMS_ENC:${aws_kms_ciphertext.keycloak_client_secret[0].ciphertext_blob}:"
+    }
+  })
+  filename = "${path.root}/${var.argocd.path}/secret-keycloak-auth.yaml"
+}
+
+locals {
+  argocd_enabled = length(var.argocd) > 0 ? 1 : 0
+  namespace      = coalescelist(var.namespace == "" && local.argocd_enabled > 0 ? [{ "metadata" = [{ "name" = var.namespace_name }] }] : kubernetes_namespace.this, [{ "metadata" = [{ "name" = var.namespace }] }])[0].metadata[0].name
+}
+
+resource "helm_release" "this" {
+  count = 1 - local.argocd_enabled
+
+  depends_on = [
+    var.module_depends_on
+  ]
+
+  name          = local.name
+  repository    = local.repository
+  chart         = local.chart
+  version       = var.chart_version
+  namespace     = local.namespace
+  recreate_pods = true
+  timeout       = 1200
+
+  dynamic "set" {
+    for_each = merge(local.conf)
+
+    content {
+      name  = set.key
+      value = set.value
+    }
+  }
+}
+
+resource "local_file" "this" {
+  count = local.argocd_enabled
+  depends_on = [
+    var.module_depends_on
+  ]
+  content  = yamlencode(local.application)
+  filename = "${path.root}/${var.argocd.path}/${local.name}.yaml"
+}
+
+
+locals {
+  name       = "kube-keycloak"
+  repository = "https://github.com/bitnami/charts/tree/master/bitnami/keycloak/"
```

Comment:
repository incorrect, this is Github, needs helm chart repo https://charts.bitnami.com/bitnami